src/main/kotlin/frc/team5499/frc2018Kotlin/utils/Vector2.kt

```diff
@@ -29,10 +29,51 @@ data class Vector2(val x: Double, val y: Double) {
 
     fun distanceTo(other: Vector2) = (this - other).magnitude
 
+    fun getHeading(): Double {
+        @Suppress("MagicNumber")
+        var rawHeading: Double = 450.0 - Math.toDegrees(Math.atan2(y, x))
+        @Suppress("MagicNumber")
+        while (rawHeading < 0.0) {
+            @Suppress("MagicNumber")
+            rawHeading += 360.0
+        }
+        @Suppress("MagicNumber")
+        while (rawHeading > 360.0) {
+            @Suppress("MagicNumber")
+            rawHeading -= 360.0
+        }
+        return rawHeading
+    }
+
     override fun toString(): String = "X: $x, Y: $y"
 
     companion object {
         val Zero = Vector2(0, 0)
+
         fun distanceBetween(a: Vector2, b: Vector2) = (a - b).magnitude
+
+        fun unitDirectionVector(vector: Vector2): Vector2 = vector * 1.0 / vector.magnitude
+
+        @Suppress("MagicNumber")
+        fun copyVector(original: Vector2): Vector2 = Vector2(original.x, original.y)
```

Comment:
you don't need this in kotlin if the class is a `data class` (https://kotlinlang.org/docs/reference/data-classes.html)

Reply:
both?